eng/template/CHANGELOG.md.tpl

```diff
@@ -0,0 +1,3 @@
+# Release History
+
+## v0.1.0 (released)
```

Comment:
Can you help me understand why this is 0.1.0? In all other languages we start at 1.0.0-beta.1 and unless there is a strong reason I'd suggest we do that in go as well.

Reply:
This might be the golang way to represent a pre-release before v1.0.0 - but I would like to defer @jhendrixMSFT to provide some more details with more accuracy.

Reply:
While golang does support a "0.x.x" versioning scheme so does a lot of other languages but we have been trying to not use that in other places as it tends to make people believe we are a far away from anything they can depend on. golang does support the prerelease versioning scheme as well according to https://golang.org/doc/modules/version-numbers#pre-release so unless there is strong reason we should be consistent with our versioning polices and go with 1.0.0-beta.1 as the first version (https://azure.github.io/azure-sdk/policies_releases.html). @jhendrixMSFT I know we talked about this a little in the past but I'd like to revisit this given we are going to be having a large influx of new libraries.